### PR TITLE
Convert codebase to use native classes

### PR DESCRIPTION
**Pull request by SergeAstapov**
Extract from #855 to keep it focused

### CODE REVIEW

File: packages/ember-blog/addon/routes/post.js [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
-    replaceWithHome() {
-      this.replaceWithExternal('home').then(() => {
-        var postController = this.controllerFor(this.routeName);
-        postController.set('replacedWithExternal', true);
-      });
-    },
-  },
-});
+  @action goToChineseVersion() {
+    this.transitionTo({ queryParams: { lang: 'Chinese' } });
```
Comment (void-mAlex) [inline packages/ember-blog/addon/routes/post.js:14]: converting this in isolation of the 855 pr change? if yes then that pr will need need to recharge this line to make tests work with e-engines-router-service due to a known bug with qp only transitions

**Comment by villander** [comment]
@SergeAstapov I don't think we need to fix it first before merging the #855 could you please clarify what you're trying to achieve here enriching the PR description?

File: packages/ember-blog/addon/routes/post.js (hunk 1 above)
Comment (SergeAstapov) [reply packages/ember-blog/addon/routes/post.js:14]: @void-mAlex sorry I think I missed your point. This PR focuses solely on syntax change, without any functional changes.

Once we land this, we we would simply need to rebase #855, do all the necessary functional tweaks without noise of syntax conversion and land it with sole focus on v5 support via using e-engines-router-service

Comment (villander) [reply packages/ember-blog/addon/routes/post.js:14]: Why do we need to merge it before the #855?

Comment (void-mAlex) [reply packages/ember-blog/addon/routes/post.js:14]: > Why do we need to merge it before the #855?

the idea is to not have a mega or that does everything



> @void-mAlex sorry I think I missed your point. This PR focuses solely on syntax change, without any functional changes.
> 
> Once we land this, we we would simply need to rebase #855, do all the necessary functional tweaks without noise of syntax conversion and land it with sole focus on v5 support via using e-engines-router-service

my concern is that a decent amount of time will be wasted debugging tests again, which is fine, it's not too daunting, but tried leaving a breadcrumb to one conflict I know we'll have

Comment (SergeAstapov) [reply packages/ember-blog/addon/routes/post.js:14]: > tried leaving a breadcrumb to one conflict I know we'll have

noted! will be glad to help

**Pull request merged**